UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -701,13 +707,8 @@ ParsePciRootBridge (
       ParsegraphicNode (Fdt, SubNode);
     }
 
-    if (AsciiStrnCmp (NodePtr->Name, "isa", AsciiStrLen ("isa")) == 0) {
-      SSubNode = FdtFirstSubnode (Fdt, SubNode); // serial
-      ParseSerialPort (Fdt, SSubNode);
-    }
-
     if (AsciiStrnCmp (NodePtr->Name, "serial@", AsciiStrLen ("serial@")) == 0) {

```

Comment:
Since the main loop now will handle both IsaSerial and PciSerial, looks like we should remove all serial handling from ParsePciRootBridge().

Reply:
I don't think I like the fact that the main loop might find itself inside the PCI node, or any other. I much prefer the approach described over here https://github.com/tianocore/edk2/pull/6382#discussion_r1821683871.

Reply:
ok. sounds good idea. in this case, the main loop only handles isa/serial and serial under root, and ParsePciRootBridge will handle isa/serial and serial under PCI, right?